### PR TITLE
HostAppointer: null host list to appoint localhost

### PR DESCRIPTION
**Pull request by matthewrmshin**
Should now work OK with no host list or explicit empty host list.

Address https://github.com/cylc/cylc/pull/2693#pullrequestreview-158071464

### CODE REVIEW

**Comment by sadielbartholomew** [comment]
Thanks for this. Tests as working for me too.

File: tests/cylc.scheduler_cli/00-host-appointer.t [hunk 1]
```diff
@@ -18,7 +18,24 @@
 # Run unit tests to test HostAppointer class for selecting hosts.
 . "$(dirname "$0")/test_header"
 #-------------------------------------------------------------------------------
-set_test_number 1
+set_test_number 3
 
 run_ok "${TEST_NAME_BASE}" python -m 'cylc.scheduler_cli'
+
+# No run hosts list
```
Comment (oliver-sanders) [inline tests/cylc.scheduler_cli/00-host-appointer.t:25]: This check is now in the unit tests, if you would prefer it here just delete the unit test https://github.com/cylc/cylc/pull/2693/commits/deb7d769b18d1f9bba4f3209dfadc67cd88fd240.

File: lib/cylc/scheduler_cli.py [hunk 2]
```diff
@@ -437,15 +435,18 @@ def _rank_good_hosts(self, all_host_stats):
                 sys.stderr.write(base_msg + ', '.join(sort_asc_hosts) + '.\n')
             return sort_asc_hosts[0]
 
-    def appoint_host(self, mock_stats=False):
+    def appoint_host(self, mock_host_stats=None):
         """Appoint the most suitable host to (re-)run a suite on."""
+        if mock_host_stats is None and (
+                not self.hosts or self.hosts == ['localhost']):
```
Comment (oliver-sanders) [inline lib/cylc/scheduler_cli.py:441]: For simplicity we can drop the `self.hosts == ['localhost']` as that logic is covered by `self._trivial_choice`.

**Comment by matthewrmshin** [comment]
Let's get the main branch merged and then refactor out any duplicate logic.

**Comment by sadielbartholomew** [comment]
Thanks @oliver-sanders & well-spotted, though as Matt suggests it will be easier now to simply change those in the follow-on PR.

**Pull request merged**